drivers/gnss/gnss_seeed_air530z.c

```diff
@@ -0,0 +1,332 @@
+	MODEM_CHAT_MATCH_WILDCARD("$??RMC,", ",*", gnss_nmea0183_match_rmc_callback),
+#if CONFIG_GNSS_SATELLITES
+	MODEM_CHAT_MATCH_WILDCARD("$??GSV,", ",*", gnss_nmea0183_match_gsv_callback),
+#endif
+);
+
+static int gnss_seeed_air530z_init_nmea0183_match(const struct device *dev)
+{
+	struct gnss_seeed_air530z_data *data = dev->data;
+
+	const struct gnss_nmea0183_match_config match_config = {
+		.gnss = dev,
+#if CONFIG_GNSS_SATELLITES
+		.satellites = data->satellites,
+		.satellites_size = ARRAY_SIZE(data->satellites),
+#endif
+	};
+
+	return gnss_nmea0183_match_init(&data->match_data, &match_config);
+}
+
+static void gnss_seeed_air530z_init_pipe(const struct device *dev)
+{
+	const struct gnss_seeed_air530z_config *config = dev->config;
+	struct gnss_seeed_air530z_data *data = dev->data;
+
+	const struct modem_backend_uart_config uart_backend_config = {
+		.uart = config->uart,
+		.receive_buf = data->uart_backend_receive_buf,
+		.receive_buf_size = sizeof(data->uart_backend_receive_buf),
+		.transmit_buf = data->uart_backend_transmit_buf,
+		.transmit_buf_size = ARRAY_SIZE(data->uart_backend_transmit_buf),
+	};
+
+	data->uart_pipe = modem_backend_uart_init(&data->uart_backend, &uart_backend_config);
+}
+
+static int gnss_seeed_air530z_init_chat(const struct device *dev)
+{
+	struct gnss_seeed_air530z_data *data = dev->data;
+
+	const struct modem_chat_config chat_config = {
+		.user_data = data,
+		.receive_buf = data->chat_receive_buf,
+		.receive_buf_size = sizeof(data->chat_receive_buf),
+		.delimiter = data->chat_delimiter,
+		.delimiter_size = ARRAY_SIZE(data->chat_delimiter),
+		.filter = NULL,
+		.filter_size = 0,
+		.argv = data->chat_argv,
+		.argv_size = ARRAY_SIZE(data->chat_argv),
+		.unsol_matches = unsol_matches,
+		.unsol_matches_size = ARRAY_SIZE(unsol_matches),
+	};
+
+	return modem_chat_init(&data->chat, &chat_config);
+}
+
+static void seeed_air530z_init_dynamic_script(const struct device *dev)
+{
+	struct gnss_seeed_air530z_data *data = dev->data;
+
+	/* Air530z doesn't response to commands. Thus, response_matches_size = 0; */
+	data->dynamic_script_chat.request = data->dynamic_request_buf;
+	data->dynamic_script_chat.response_matches = NULL;
+	data->dynamic_script_chat.response_matches_size = 0;
+	data->dynamic_script_chat.timeout = 0;
+
+	data->dynamic_script.name = "PCAS";
+	data->dynamic_script.script_chats = &data->dynamic_script_chat;
+	data->dynamic_script.script_chats_size = 1;
+	data->dynamic_script.abort_matches = NULL;
+	data->dynamic_script.abort_matches_size = 0;
+	data->dynamic_script.callback = NULL;
+	data->dynamic_script.timeout = 5;
+}
+
+static int gnss_seeed_air530z_init(const struct device *dev)
+{
+	struct gnss_seeed_air530z_data *data = dev->data;
+	const struct gnss_seeed_air530z_config *config = dev->config;
+	int ret;
+
+	LOG_DBG("initializing SEED Air530z");
+
+	ret = gnss_seeed_air530z_init_nmea0183_match(dev);
+	if (ret < 0) {
+		LOG_ERR("ERROR gnss_seeed_air530z_init_nmea0183_match");
```

Comment:
Is this log really needed?

Reply:
Actually no, I added it during development and I forgot to remove it. Done!